### PR TITLE
update zhipuai llm and fix issues

### PR DESCRIPTION
**Pull request by nightosong**
# Description

Here are the updates:

* **Name Change ZhipuAI -> ZhipuAILLM:** Since there are also other models like Embedding and MultiModal, ZhipuAILLM is more appropriate in this context than ZhipuAI.
* **Updated Unit Tests and Bug Fixes.**

Fixes # (issue)

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [X] Yes

## Type of Change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [X] I added new unit tests to cover this change
- [X] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by nightosong** [comment]
@nerdai could you pls take a look at this pr.

File: llama-index-integrations/llms/llama-index-llms-zhipuai/README.md [hunk 1]
```diff
@@ -11,13 +11,13 @@
 
 ```py
 # Import ZhipuAI
-from llama_index.llms.zhipuai import ZhipuAI
+from llama_index.llms.zhipuai import ZhipuAILLM
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-zhipuai/README.md:14]: I think the convention that most have adopted would be to use what you have already:
- `ZiphuAI` for the LLM and `ZiphuAIEmbedding` for the embedding.

Comment (nerdai) [reply llama-index-integrations/llms/llama-index-llms-zhipuai/README.md:14]: For example:

- for LLMs: `Cohere`, `OpenAI`, `AlephAleph`, `Databricks` 
- for embeddings: `CohereEmbedding`, `OpenAIEmbedding`, `AlephAlephEmbedding`, and `DataBricksEmbedding`

However, with that being said there are a minority of cases that use the naming convention you are suggesting to switch to here. I'll leave it up to you.

**Review by nerdai — APPROVED** [review]
Thanks @nightosong. I left a comment re: the naming switch of the class, but other than that it looks good.

File: llama-index-integrations/llms/llama-index-llms-zhipuai/README.md (hunk 1 above)
Comment (nightosong) [reply llama-index-integrations/llms/llama-index-llms-zhipuai/README.md]: thanks for your reminder and advice.

**Review by nerdai — APPROVED** [review]

**Pull request merged**